sdk/src/main/java/com/mapbox/maps/MapProvider.kt

```diff
@@ -50,6 +53,12 @@ internal object MapProvider {
     return mapTelemetry
   }
 
+  fun getEventService(accessToken: String): EventsService {
+    val eventsServiceOptions =
+      EventsServerOptions(accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, null)
```

Comment:
If the token is passed from outside - maybe pass useragent as well?

Reply:
If we expose both, I guess no need to wrap to this object class 😅 The `MAPBOX_EVENTS_USER_AGENT` is not supposed to change with each build, the access token can be runtime-configured on the other hand.

Reply:
As far as I understood this class - it just abstracts creation of objects, that eases testing =) But anyway - not critical, up to you to decide

Reply:
refactored to flushEvents as suggested by @jush